src/authorization/OwnerPermissionReader.ts

```diff
@@ -24,33 +27,54 @@ export class OwnerPermissionReader extends PermissionReader {
     this.aclStrategy = aclStrategy;
   }
 
-  public async handle(input: PermissionReaderInput): Promise<PermissionSet> {
+  public async handle(input: PermissionReaderInput): Promise<PermissionMap> {
+    const result: PermissionMap = new IdentifierMap();
+    const acls = [ ...this.filterAcls(input.accessMap.keys()) ];
+    if (acls.length === 0) {
+      this.logger.debug(`No ACL resources found that need an ownership check.`);
+      return result;
+    }
+
+    let podBaseUrl: string;
     try {
-      await this.ensurePodOwner(input);
+      podBaseUrl = await this.findPodBaseUrl(input.credentials);
```

Comment:
> That's not going to work with multiple pods, is it?

This class will have to change once that is supported yes.

Not sure what the problem is with the check? The credentials are the same for every resource, so we only have to find the pod URL once. And then we have to check which resources are contained within that pod URL.

Reply:
Once this change is here we would just get an array of pod URLs and the loop would change to check for each identifier if it contains one of the entries in the array instead of just the single instance we have now.

Reply:
Okay; let's have an issue to create an injectable object (`PodStrategy`?) with an `.isOwnerOf` or so method. There's assumptions being hardcoded here.